envpool/sokoban/level_loader.cc

```diff
@@ -183,15 +188,15 @@ std::vector<SokobanLevel>::iterator LevelLoader::GetLevel(std::mt19937& gen) {
   if (n_levels_to_load_ > 0 && levels_loaded_ >= n_levels_to_load_) {
     throw std::runtime_error("Loaded all requested levels.");
   }
-  if (cur_level_ == levels_.end()) {
+  while (cur_level_ >= levels_.size()) {
+    cur_level_ -= levels_.size();
     LoadFile(gen);
-    cur_level_ = levels_.begin();
-    if (cur_level_ == levels_.end()) {
+    if (levels_.size() == 0) {
       throw std::runtime_error("No levels loaded.");
     }
   }
-  auto out = cur_level_;
-  cur_level_++;
+  auto out = levels_.begin() + cur_level_;
```

Comment:
This is undefined behavior (https://stackoverflow.com/a/1057788/4521118) because we haven't checked that `cur_level_ < levels_.size()`. I would advise checking for that instead of `levels_.size() == 0`.

Reply:
We are check it in the condition of the while loop. When the while loop terminates, `cur_level_` will be less than `levels_.size()`